src/Telemetry/AzureDevOpsTimeline/BuildLogScraper.cs

```diff
@@ -21,24 +20,23 @@ public class BuildLogScraper : IBuildLogScraper
         // Example: mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7-3e800f1-20190501005343
         private static readonly Regex _dockerImageRegex = new Regex(@"mcr.microsoft.com\/dotnet-buildtools\/prereqs:\S+");
 
-        public BuildLogScraper(ILogger<BuildLogScraper> logger, IAzureDevOpsClient client)
+        public BuildLogScraper(ILogger<BuildLogScraper> logger)
         {
-            _azureDevOpsClient = client;
             _logger = logger;
         }
 
-        public Task<string> ExtractMicrosoftHostedPoolImageNameAsync(string logUri, CancellationToken cancellationToken)
-            => ExtractImageNameAsync(logUri, ExtractMicrosoftHostedImageName, cancellationToken);
+        public Task<string> ExtractMicrosoftHostedPoolImageNameAsync(IAzureDevOpsClient client, string logUri, CancellationToken cancellationToken)
+            => ExtractImageNameAsync(client, logUri, ExtractMicrosoftHostedImageName, cancellationToken);
 
-        public Task<string> ExtractOneESHostedPoolImageNameAsync(string logUri, CancellationToken cancellationToken)
-            => ExtractImageNameAsync(logUri, ExtractOneESHostedImageName, cancellationToken);
+        public Task<string> ExtractOneESHostedPoolImageNameAsync(IAzureDevOpsClient client, string logUri, CancellationToken cancellationToken)
```

Comment:
Hrm... this feels like bad DI here.  Why should I have to pass you a configured client? Isn't that the point of DI?  It's odd the caller needs to make those distinctions.  It feels like this could have an IOptions and that tells it what project (or projects) to read and use that.

Reply:
I guess... this could pass in what org to talk to? That feels strange though because we are already talking to that org, and already made a client. It may be possible to extract "what org" from the logUri, but I have no idea what the format of that uri is, and it will probably change and break whatever we write to do that.